src/mon/FSCommands.cc

```diff
@@ -241,6 +241,18 @@ class FsNewHandler : public FileSystemCommandHandler
       }
     }
 
+    int64_t fscid = FS_CLUSTER_ID_NONE;
+    if (cmd_getval(cmdmap, "fs_id", fscid)) {
+      if (!force) {
+        ss << "Pass --force to create a file system with a specific ID";
+        return -EINVAL;
+      }
+      if (fsmap.filesystem_exists(fscid)) {
+        ss << "filesystem already exists with id '" << fscid << "'";
+        return -EINVAL;
```

Comment:
I saw you added a test for idempotency but shouldn't this check fail it?

Reply:
No it won't. If the FS name and  pools are already in use, then the command returns earlier with no-op. Please see, https://github.com/ceph/ceph/blob/master/src/mon/FSCommands.cc#L198

Reply:
Got it, thanks.